### PR TITLE
FindReplaceLogic.updateTarget should call endSession on current target

### PR DESCRIPTION
**Pull request by merks**

### CODE REVIEW

File: bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/FindReplaceLogic.java [hunk 1]
```diff
@@ -588,12 +588,11 @@ public void updateTarget(IFindReplaceTarget newTarget, boolean canEditTarget) {
 		this.isTargetEditable = canEditTarget;
 
 		if (this.target != newTarget) {
-			if (newTarget != null && newTarget instanceof IFindReplaceTargetExtension)
-				((IFindReplaceTargetExtension) newTarget).endSession();
+			if (this.target instanceof IFindReplaceTargetExtension)
```
Comment (merks) [inline bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/FindReplaceLogic.java:591]: We should end the session for the current target not for the new target.

File: bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/FindReplaceLogic.java [hunk 2]
```diff
@@ -588,12 +588,11 @@ public void updateTarget(IFindReplaceTarget newTarget, boolean canEditTarget) {
...
+				((IFindReplaceTargetExtension) this.target).endSession();
 
 			this.target = newTarget;
-			if (newTarget != null)
-				isTargetSupportingRegEx = newTarget instanceof IFindReplaceTargetExtension3;
+			isTargetSupportingRegEx = newTarget instanceof IFindReplaceTargetExtension3;
```
Comment (merks) [inline bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/FindReplaceLogic.java:595]: The null guard seems pointless.

**Comment by merks** [comment]
@Wittmaxi 

FYI, I noticed that find in Oomph's setup editors was broken, often requiring me to close the editor to get it working again.   I tested against EMF's editors (e.g., the Ecore editor) and it had the same problem.  I finally tracked it down to this lifecycle problem.

EMF add this control:

image

And when the editor no longer has focus, the session should end and the control should be removed:

image

File: bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/FindReplaceLogic.java (hunk 1 above)
Comment (HeikoKlare) [reply bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/FindReplaceLogic.java:591]: Thank you for finding this, @merks! This is a regression from #1132. Your proposal properly restored the previous behavior.

**Review by HeikoKlare — APPROVED** [review]
Thank you for the fix! It properly corrects a recently introduced regression. Thus, I propose to merge this once CI has finished. Then it should also make its way into the M2 build.

**Comment by merks** [comment]
Windows/Mac failures unrelated.

**Pull request merged**